aip/0231.md

```diff
@@ -0,0 +1,86 @@
+---
+aip:
+  id: 231
+  state: draft
+  created: 2019-06-18
+permalink: /231
+redirect_from:
+  - /0231
+---
+
+# Batch Get
+
+Some APIs need to allow users to get a specific set of resources at a consistent
+time point (e.g. using a read transaction). A batch get method provides this
+functionality.
+
+## Guidance
+
+Batch get methods are specified using the following pattern:
+
+```proto
+rpc BatchGetBooks(BatchGetBooksRequest) returns (BatchGetBooksResponse) {
+  option (google.api.http) = {
+    get: "/v1/shelves/-/books:batchGet"
```

Comment:
```suggestion
    get: "/v1/shelves/*/books:batchGet"
```

...because the stuff below says that `-` may be supported, but this makes it the only option.

Reply:
Question: if the http option is specified with a star, but there is no variable in the http option, is there any way for the API to verify that the URL path matches the `name` field in every request?

Reply:
Oh, right. I assert there needs to be a variable in the http option (and corresponding field in the proto).

Reply:
I discussed this with @jgeewax and we believe that it should always be a `*` and include a variable name. We should permit APIs to use `-`, but not require them to support it, and it should not be the "sensible default".

Reply:
This will be resolved by recommending a `parent` field.